pkg/apis/extensions/types.go

```diff
@@ -437,6 +449,22 @@ type RollingUpdateDaemonSet struct {
 	MaxUnavailable intstr.IntOrString
 }
 
+// Spec to control the desired behavior of a daemon set surging rolling update.
+type SurgingRollingUpdateDaemonSet struct {
+	// The maximum number of DaemonSet pods that can be scheduled above the desired number of pods
+	// during the update. Value can be an absolute number (ex: 5) or a percentage of the total number
+	// of DaemonSet pods at the start of the update (ex: 10%). The absolute number is calculated from
+	// the percentage by rounding up. This cannot be 0. The default value is 1. Example: when this is
+	// set to 30%, at most 30% of the total number of nodes that should be running the daemon pod
+	// (i.e. status.desiredNumberScheduled) can have 2 pods running at any given time. The update
```

Comment:
This may need to be reworded, it is a little confusing:

```
Example: when this is set to 30%, at most 30% of the total number of nodes that
should be running the daemon pod (i.e. status.desiredNumberScheduled) can have
2 pods running at any given time...
```

Where does the `2 pods running` calculation come from?